main_test.go

```diff
@@ -588,6 +642,14 @@ func startHTTP() (net.Listener, chan struct{}) {
 	return ln, done
 }
 
+func startRedis(t *testing.T) *miniredis.Miniredis {
+	redis := miniredis.NewMiniRedis()
+	if err := redis.StartAddr("127.0.0.1:6379"); err != nil {
+		t.Fatalf("Failed to create redis server: %s", err)
+	}
+	return redis
+}
+
 func startCHServer() {
 	http.ListenAndServe(":8124", http.HandlerFunc(fakeCHHandler))
```

Comment:
Same thing here

Reply:
This one is more complex because it requires additional change in configurations (that how we define destination port of Clickhouse). This one is just a mock. I'll leave a todo, to be handled in a separate PR